### PR TITLE
feat(components/datetime): add aria labels to datepicker calendar navigation buttons

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2599956](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2599956)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2033/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2033/storybooks/ag-grid-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2033/storybooks/datetime-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2033/storybooks/lists-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2033/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2033/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2033/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2033/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2033/playground/)

File: libs/components/datetime/src/assets/locales/resources_en_US.json [hunk 1]
```diff
@@ -7,6 +7,30 @@
...
+  "skyux_datepicker_move_calendar_next_month": {
+    "_description": "ARIA label for the next month calendar button in 'day' mode",
+    "message": "Next year"
+  },
+  "skyux_datepicker_move_calendar_previous_year": {
+    "_description": "ARIA label for the previous month calendar button in 'day' mode",
+    "message": "Previous page"
+  },
+  "skyux_datepicker_move_calendar_next_year": {
+    "_description": "ARIA label for the next month calendar button in 'day' mode",
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/assets/locales/resources_en_US.json]: These descriptions all mention "month", but their keys say `year`, `month`, and `day`. Can the descriptions be tweaked a bit to better align with their keys?

File: libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts [hunk 2]
```diff
@@ -129,16 +134,43 @@ export class SkyDatepickerCalendarInnerComponent
...
+  #nextDay: string | undefined;
+  #prevMonth: string | undefined;
+  #nextMonth: string | undefined;
+  #prevYear: string | undefined;
+  #nextYear: string | undefined;
 
   #_selectedDate: Date | undefined;
   #_startingDay = 0;
 
+  #resourcesSvc = inject(SkyLibResourcesService);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts]: Needs `readonly` for injected properties.

File: libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts [hunk 3]
```diff
@@ -129,16 +134,43 @@ export class SkyDatepickerCalendarInnerComponent
...
+    this.#resourcesSvc
+      .getStrings({
+        prevDay: 'skyux_datepicker_move_calendar_previous_day',
+        nextDay: 'skyux_datepicker_move_calendar_next_day',
+        prevMonth: 'skyux_datepicker_move_calendar_previous_month',
+        nextMonth: 'skyux_datepicker_move_calendar_next_month',
+        prevYear: 'skyux_datepicker_move_calendar_previous_year',
+        nextYear: 'skyux_datepicker_move_calendar_next_year',
+      })
+      .pipe(take(1))
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts]: Using `take` is convenient, but it prevents the subscription from responding to changes to the resources service `getStrings` method. For example, a new value will be emitted from `getStrings` when the locale changes.

File: libs/components/datetime/src/assets/locales/resources_en_US.json (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/datetime/src/assets/locales/resources_en_US.json]: Fixed

File: libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts]: Added

File: libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts]: Updated to takeUntil unsubscribe

File: libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts [hunk 4]
```diff
@@ -129,16 +134,43 @@ export class SkyDatepickerCalendarInnerComponent
...
+        nextYear: 'skyux_datepicker_move_calendar_next_year',
+      })
+      .pipe(takeUntil(this.#ngUnsubscribe))
+      .subscribe((resources) => {
+        this.#prevDay = resources.prevDay;
+        this.#nextDay = resources.nextDay;
+        this.#prevMonth = resources.prevMonth;
+        this.#nextMonth = resources.nextMonth;
+        this.#prevYear = resources.prevYear;
+        this.#nextYear = resources.nextYear;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts:172]: Won't the view need to be refreshed when new strings are returned from the resources service? There are a few async behaviors in this component that could affect the view; maybe rethink how they should all interact when one or more of them change?

Comment (Blackbaud-CoreyArcher) [reply libs/components/datetime/src/lib/modules/datepicker/datepicker-calendar-inner.component.ts:172]: I think just adding a call to `this.refreshView()` should handle everything. On calendar instantiation, it won't run if the child view hasn't been initalized, but then everything other change detection cycle runs through it so these resource strings should too.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/2042

**Pull request merged**